src/controllers/timeTaskCtrl.js

```diff
@@ -0,0 +1,204 @@
+      // else verify process for start a new timer
+      if (data.message) {
+        const newTimeTask = new TimeTask({ task: taskId, time: {} });
+        newTimeTask
+          .save()
+          .then(() => res.status(201).json({ message: 'Timer created', newTimeTask }))
+          .catch((error) => res.status(500).json({ error }));
+      } else {
+        let lastTime = data.timeTask.time[data.timeTask.time.length - 1];
+        // If end_date is null so a timer is running
+        if (!(lastTime.end_date)) {
+          return res.status(400).json({ message: 'A timer is running on this task, end the timer and retry' })
+        }
+
+        // Else init a new timer
+        const updTime = {
+          time: [
+            ...data.timeTask.time,
+            {
+              start_date: new Date().toISOString()
+            }
+          ]
+        }
+
+        // Update TimeTask with the new timer
+        TimeTask.findByIdAndUpdate(data.timeTask.id, updTime, { new: true })
+          .then((result) => {
+            if (!result) {
+              return res.status(404).json({ message: 'Timer not found' });
+            }
+            return res
+              .status(200)
+              .json({ message: 'The timer has been modified correclty', result });
+          })
+          .catch((error) => res.status(400).json({ error }));
+      }
+    })
+    .catch(error => res.status(400).json({ error }))
+};
+
+exports.endTimeTask = (req, res) => {
+  getTimeTask(req.params.task_id)
+    .then((data) => {
+      // data.message mean timeTask not found, so return error
+      if (data.message) {
+        return res.status(404).json({ message: 'Timer not found, start a new timer for init' });
+      }
+
+      let lastTime = data.timeTask.time[data.timeTask.time.length - 1];
+      // If end_date is true, no timer is running and return error message
+      if (lastTime.end_date) {
+        return res.status(400).json({ message: 'No timer started' })
+      }
+
+      // Else stop current timer is running
+      lastTime.end_date = new Date().toISOString();
+      let updTime = {
+        time: [
+          ...data.timeTask.time
+        ]
+      }
+
+      TimeTask.findByIdAndUpdate(data.timeTask.id, updTime, { new: true })
+        .then((result) => {
+          if (!result) {
+            return res.status(404).json({ message: 'Timer not found' });
+          }
+          return res
+            .status(200)
+            .json({ message: 'The timer has been modified correclty', result });
+        })
+        .catch((error) => res.status(400).json({ error }));
+    })
+    .catch((error) => res.status(400).json({ error }));
+};
+
+exports.getUserTimeTasks = (_, res) => {
+  Task.find({ assigned: res.locals.user.id })
+    .then(async (task) => {
+      if (!task) {
+        return res.status(404).json({ message: 'Task not found' });
+      }
+      // Find timer data of all Task found
+      let timeTaskFound = [];
```

Comment:
In my mind we can't have 2 tasks running, when we run another task a message say `Task already running, please end it or pause running task`. What is your opinion?